### PR TITLE
Parametric project data

### PR DESCRIPTION
**Pull request by akey7**
This is for the parametric project data in dataframes.

### CODE REVIEW

File: landbosse/excelio/XlsxManagerRunner.py [hunk 1]
```diff
@@ -107,3 +117,70 @@ def extract_details_lists(self, runs_dict):
...
+        parametrics is returned that has all the columns but no rows so
+        it can be used in joins. It doe snot matter what the sheet name
+        is.
+
+        2. Project list spreadsheet contains two sheets: The sheet named
+        "Project list" is assumed to be the project list and the sheet
+        named "Parametric list" is assumed to be the specifications for
+        the parametric variables. If one or both of those tab names are
+        missing, an exception is raised
+
```
Comment (eberlea) [inline landbosse/excelio/XlsxManagerRunner.py]: I don't think the back-compatibility is working correctly right now. I get the following error and I think the root of the problem is in this method. 
image

File: landbosse/excelio/XlsxManagerRunner.py [hunk 2]
```diff
@@ -107,3 +117,70 @@ def extract_details_lists(self, runs_dict):
...
+        else:
+            raise KeyError("Project list needs to have a single sheet or sheets named 'Project list' and 'Parametric list'.")
+
+        # Instantiate and XlsxReader to assemble master input dictionary
+        xlsx_reader = XlsxReader()
+
+        # Join in the parametric variable modifications
+        parametric_value_list = xlsx_reader.create_parametric_value_list(parametric_list)
+        enhanced_project_list = xlsx_reader.outer_join_projects_to_parametric_values(project_list,
+                                                                                 parametric_value_list)
```
Comment (eberlea) [inline landbosse/excelio/XlsxManagerRunner.py]: It looks like the parametric_value_list is empty when it gets called here using the old data so it's probably a problem with the xlsx_reader.create_parametric_value_list function. Would you review and revise? I can try to fix it myself as well but I don't have time this evening so I thought I'd see if you happen to have time today. No worries if not, I'll resume my review on Monday either way and can make revisions then if needed.

File: landbosse/excelio/XlsxManagerRunner.py (hunk 1 above)
Comment (akey7) [reply landbosse/excelio/XlsxManagerRunner.py]: I found this problem about the same time you did. I fixed it and re-ran validation using data compatible with version 2.1.5 and it works now.

**Comment by akey7** [comment]
The backward compatibility should be fixed now.

**Comment by akey7** [comment]
1. File structure has been updated

1. Validation now fails and passes as expected.

1. `Project ID with serial` and `project_id_with_serial` usage is now consistent throughout the code.

**Pull request merged**